src/lib/storage/table.hpp

```diff
@@ -225,7 +227,16 @@ class Table : private Noncopyable {
   const std::vector<ColumnID>& value_clustered_by() const;
   void set_value_clustered_by(const std::vector<ColumnID>& value_clustered_by);
 
+  /*
+   * Persist table to use mmap-based storage for its data.
+   * TODO: Explanation.
+   */
+
+  void persist();
+
  protected:
+  void _persist_chunk(ChunkID chunk_id);
```

Comment:
Do we have a use case that requires any component to persist/snapshot an individual chunk? Assuming we persist a table at time t1 and later at t2 again since the table has more chunks at t2, we want to have the functionality of persisting individual chunks that persist the delta instead of writing all chunks again. However, this could also be implemented as an optimization of `persist_table`. Then, we would call `persist_table` at t1 and t2, but at the second call, only the delta of immutable chunks will be persisted.

Reply:
For better object orientation, I think the chunks should be able to persist themselves.

Reply:
I wouldn't argue that this is better object orientation. Apart from that: OO is soooooo 2000s! :D